docs/run-instance/installation/building-from-source.md

```diff
@@ -0,0 +1,150 @@
+# Building from Source
+
+If you need to customize your openrouteservice instance even further than what is possible by [configuration](/run-instance/configuration/), or want to start contributing to the openrouteservice project, the following section will give you starting points.
+
+## Prerequisites
+
+The following documentation assumes you are running an Ubuntu 20.04 system (also generally works with newer Ubuntu versions). Depending on your environent, you might need to adjust certain details. You will also need to make sure to have the following installed: 
+* [git](https://github.com/git-guides/install-git) should be available on your system
+* [java](https://www.java.com/en/) 17 should be available, preferably as default Java environment
+* [maven](https://maven.apache.org/) should be installed on your system
+
+## Installation from source
+
+To build openrouteservice from source, you can use the following commands:
+```shell
+git clone https://github.com/user/openrouteservice.git
+cd openrouteservice
+mvn package
+```
+
+If you have made modifications to the code, you should run all tests before building and using openrouteservice. For a significant part of the tests you need to activate a maven profile called `apitests` to run them. Use the following commands:
+
+```shell
+mvn -Papitests verify
+# afterwards you can build the artefacts without running the tests again by issuing the following command:
+mvn -DskipTests package
+```
+
+[//]: # (TODO: overhaul contents below after integrating the jar build PR)
+After running this command, you will find the artefact at `ors-api/target/ors.war`. 
+
+After you have packaged openrouteservice, there are two options for running it. One is to run the `mvn spring-boot:run` command which triggers a spring-boot native Tomcat instance running on port `8082`.  This is more restrictive in terms of settings for Tomcat. The other is to install and run Tomcat 10. In both cases the requirements of configuration and source files discussed [here](running-jar-war) apply.
+
+[//]: # (TODO: the part below partly belongs somewhere in the contributing section; running with IDE needs to be overhauled, too, since it does not mention running with spring boot run config)
+
+## Running from within IDE
+
+To run the project from within your IDE, you have to:
+
+  1. Set up your IDE project and import `openrouteservice`
+     modules as Maven model.
+     For IntelliJ Idea, have a look at [these instructions](/contributing/opening-project-in-intellij).
+
```

Comment:
When using the Intellij generated "spring boot Application" run configuration. This doesn't work out of the box for me. I needed to set the Working directory to <repo_root>/ors-api for the default config paths to work.

The current paths in ors-config.yml also seem to be done for the docker setup only. As i now get ors-core folders spawned, wherever the current working directory is set (root or ors-api folder), which are also not ignored?

Reply:
Things that should eventually work out of the box:

Intellij run configs for:
- autogenerated Spring Boot task
- maven `spring-boot:run` task (on openrouteservice or ors-api folder?)
(maybe provide some info on what you normally run when developing ORS

Command line:
`mvn spring-boot:run`
`docker compose up -d`

Reply:
jar / war & docker setup will be tackled outside this PR